snapshot_dbg_cli_tests/test_cli_services.py

```diff
@@ -0,0 +1,540 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" Unit test file for the CliServices class.
+"""
+
+import argparse
+import unittest
+
+from snapshot_dbg_cli.cli_services import CliServices
+from snapshot_dbg_cli.cli_services import CHECK_CONFIGURATION_MSG
+from snapshot_dbg_cli.exceptions import SilentlyExitError
+from snapshot_dbg_cli.firebase_types import FirebaseProject
+from snapshot_dbg_cli.firebase_types import FirebaseProjectGetResponse
+from snapshot_dbg_cli.firebase_types import FirebaseProjectStatus
+from snapshot_dbg_cli.firebase_types import DatabaseGetResponse
+from snapshot_dbg_cli.firebase_types import DatabaseGetStatus
+from snapshot_dbg_cli.firebase_types import DatabaseInstance
+from unittest.mock import ANY
+from unittest.mock import call
+from unittest.mock import MagicMock
+from unittest.mock import patch
+
+
+def build_firebase_project(defaut_rtdb_name=None):
+  project_data = {'resources': {}}
+
+  if defaut_rtdb_name is not None:
+    project_data['resources']['realtimeDatabaseInstance'] = defaut_rtdb_name
+
+  return FirebaseProject(project_data)
+
+
+class CliServicesTests(unittest.TestCase):
```

Comment:
In general, with this set of tests, I'm unclear on how many of the assertions are testing the implementation vs the functionality, nor where that distinction is important.  Happy to discuss.

Reply:
I've made some changes and added some comments. PTAL.